fga/cache/cachedFgaTenantStore.go

```diff
@@ -0,0 +1,88 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/hashicorp/golang-lru/v2/expirable"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc/status"
+)
+
+type CachedFgaTenantStore struct {
+	cache *expirable.LRU[string, string]
+}
+
+func NewCachedFgaHelper() *CachedFgaTenantStore {
+	return &CachedFgaTenantStore{
+		cache: expirable.NewLRU[string, string](10, nil, 10*time.Minute),
+	}
+}
+
+func (c *CachedFgaTenantStore) GetStoreIDForTenant(ctx context.Context, conn openfgav1.OpenFGAServiceClient, tenantID string) (string, error) {
```

Comment:
Lets introduce the interface to each method, so we can mock it if needed.

Reply:
done